examples/python/get_open_access_pdf/parallel.py

```diff
@@ -77,11 +77,10 @@ async def download_papers(paper_ids: list[str], directory: str = 'papers', batch
             if not paper['isOpenAccess']:
                 yield paper_id, None
 
-            paperId: str = paper['paperId']
-            pdf_url: str = paper['openAccessPdf']['url']
-            pdf_path = os.path.join(directory, f'{paperId}.pdf')
-
             try:
+                paperId: str = paper['paperId']
+                pdf_url: str = paper['openAccessPdf']['url']
```

Comment:
I don't know what exception is thrown by `download_pdf` when the url is `None`, but you could probably get a more clear error message by doing something like:
```python
try:
                paperId: str = paper['paperId']
                pdf_url: str = paper['openAccessPdf']['url']
                if pdf_url is None:
                    raise Exception("No PDF for this paper")
                pdf_path = os.path.join(directory, f'{paperId}.pdf')
                await download_pdf(session, pdf_url, pdf_path)
                yield paper_id, pdf_path
```

Perhaps that's too fussy for this example though. Up to you.